From bf835c49ae2ead6f3c46bda3cd9fa1e268ae4940 Mon Sep 17 00:00:00 2001
From: Petri Hintukainen <phintuka@gmail.com>
Date: Wed, 5 Aug 2026 18:52:45 +0300
Subject: [PATCH] m2ts_demux: Fix possible buffer overread when parsig PTS/DTS

Overread possible if ts payload_offset is near packet end.
---
 src/libbluray/decoders/m2ts_demux.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/src/libbluray/decoders/m2ts_demux.c b/src/libbluray/decoders/m2ts_demux.c
index 73bd4d9d..1319a554 100644
--- a/src/libbluray/decoders/m2ts_demux.c
+++ b/src/libbluray/decoders/m2ts_demux.c
@@ -180,9 +180,17 @@ static int _parse_pes(PES_BUFFER *p, uint8_t *buf, unsigned len)
         }
 
         if (pts_exists) {
+            if (hdr_len < 14) {
+                BD_DEBUG(DBG_DECODE, "invalid BDAV TS (PES header too short for PTS)\n");
+                return -1;
+            }
             p->pts = _parse_timestamp(buf + 9);
         }
         if (dts_exists) {
+            if (hdr_len < 19) {
+                BD_DEBUG(DBG_DECODE, "invalid BDAV TS (PES header too short for DTS)\n");
+                return -1;
+            }
             p->dts = _parse_timestamp(buf + 14);
         }
     }
